### PR TITLE
Merge docker into main

### PR DESCRIPTION
**Pull request by jamie-meyer**
Docker branch contains the latest changes for the fireant program. The main branch should actually contain all of these changes, but we got stuck with making changes on docker branch. Should be ported for consistency and maintainability.

### CODE REVIEW

File: main.py [hunk 1]
```diff
@@ -42,19 +49,194 @@ def setup_env():
...
+    resp = json.loads(resp.data)
+
+    # Ensure the version is stable (e.g., alpha or beta not in version)
+    i = 0
+    while any(char.isalpha() for char in resp['response']['docs'][i]['v']):
+        i += 1
+    version = resp['response']['docs'][i]['v']
+
+    return version
+
```
Comment (lewismc) [inline main.py:68]: Please remove all whitespace.

**Review by lewismc — CHANGES_REQUESTED** [review]
Thanks @jamie-meyer and anyone else who contributed to this. I think we should remove whitespace and then merge. We can then go back and improve with documentation. We can also run deepsource on the code and fix any issues identified there.

File: main.py (hunk 1 above)
Comment (jamie-meyer) [reply main.py:63]: All whitespace has been removed.

**Comment by lewismc** [comment]
@jamie-meyer thanks for updating the PR. When I said please remove whitespace I should have asked you to apply this to the entire pull request. Excess whitespace exists all over the place. Please remove it where you can. Thanks

**Comment by lewismc** [comment]
actually @jamie-meyer please just create another issue to address the whitespace issue. Thanks

**Pull request merged**